src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-filter/GeneFilter.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { connect } from 'react-redux';
+import { RootState } from 'src/store';
+import { isEntityViewerSidebarOpen } from '../../../state/sidebar/entityViewerSidebarSelectors';
+
+import { ReactComponent as CloseIcon } from 'static/img/shared/close.svg';
+import chevronUpIcon from 'static/img/shared/chevron-up.svg';
+import RadioGroup, {
+  RadioOptions
+} from 'src/shared/components/radio-group/RadioGroup';
+
+import styles from './GeneFilter.scss';
+import Checkbox from 'src/shared/components/checkbox/Checkbox';
+
+type Props = {
+  toggleFilter : () => void ;
+  isSidebarOpen : boolean;
+};
+
+const radioData: RadioOptions = [
+  { value: 'default', label: 'Default' },
+  { value: 'length_longest', label: 'Spliced length: longest - shortest' },
+  { value: 'length_shortest', label: 'Spliced length: shortest - longest' }
+];
+
+const GeneFilter = (props: Props) => {
+
+  const [isChecked, setChecked] = useState(false);
+
+  const filterBoxClassnames = classNames(
+    styles.filterBox,
+    { [styles.filterBoxFullWidth]: !props.isSidebarOpen }
+  );
+
+  const [selectedRadio, setselectedRadio] = useState('default');
+
+  const radioChange = (value: string) => {
```

Comment:
@ens-ma7 The issues is here. The type of `value` should be `OptionValue` instead of `string`.

Reply:
Thanks @imransl i copied this code from the storybook example for radiogroup:
https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/stories/shared-components/radio-group/RadioGroup.stories.tsx#L34

How is it working there but not in my code.

Reply:
It probably should've highlighted the error there too. However, in this case the deployment requires to check the types in the `src` code. So it caught this error, which resulted in the deployment stopping half way.